osu.Server.Queues.ScorePump/ImportHighScores.cs

```diff
@@ -0,0 +1,138 @@
+                string query = $"SELECT * FROM {highScoreTable} WHERE score_id >= @startId";
+
+                Console.WriteLine($"Querying with \"{query}\"");
+
+                var highScores = dbMainQuery.Query<HighScore>(query, new { startId = StartId }, buffered: false);
+
+                foreach (var highScore in highScores)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                        break;
+
+                    Console.WriteLine($"Reading score {highScore.score_id}");
+
+                    // Only used to calculate accuracy and statistics.
+                    var scoreInfo = new ScoreInfo
+                    {
+                        Ruleset = ruleset.RulesetInfo,
+                        RulesetID = RulesetId,
+                    };
+                    scoreInfo.SetCount50(highScore.count50);
+                    scoreInfo.SetCount100(highScore.count100);
+                    scoreInfo.SetCount300(highScore.count300);
+                    scoreInfo.SetCountMiss(highScore.countmiss);
+                    scoreInfo.SetCountGeki(highScore.countgeki);
+                    scoreInfo.SetCountKatu(highScore.countkatu);
+                    LegacyScoreDecoder.PopulateAccuracy(scoreInfo);
+
+                    // Convert to new score format
+                    var soloScore = new SoloScore
+                    {
+                        user_id = highScore.user_id,
+                        beatmap_id = highScore.beatmap_id,
+                        ruleset_id = RulesetId,
+                        preserve = true,
+                        ScoreInfo = new SoloScoreInfo
+                        {
+                            id = highScore.score_id,
+                            user_id = highScore.user_id,
+                            beatmap_id = highScore.beatmap_id,
+                            ruleset_id = RulesetId,
+                            passed = true,
+                            total_score = highScore.score,
+                            accuracy = scoreInfo.Accuracy,
+                            max_combo = highScore.maxcombo,
+                            rank = highScore.rank,
+                            mods = ruleset.ConvertFromLegacyMods((LegacyMods)highScore.enabled_mods).Select(m => new APIMod(m)).ToList(),
+                            statistics = scoreInfo.Statistics,
+                            started_at = highScore.date,
+                            ended_at = highScore.date,
+                            created_at = highScore.date,
+                            updated_at = highScore.date
+                        },
+                        created_at = highScore.date,
+                        updated_at = highScore.date,
+                    };
+
+                    // Todo: Import highScore.hidden somehow?
+
+                    using (var transaction = db.BeginTransaction())
```

Comment:
Definitely don't want one transaction per insert, but can fix that when we get closer to running this on mass scores.

Reply:
One thing to note is that currently the previous score id is stored inside the json data. So right now `score.score_id != score.data.score_id` for these particular scores.